File: reactivando-backend/src/main/java/com/reactivando/futuro/service/AuditoriaService.java
package com.reactivando.futuro.service;

import com.reactivando.futuro.dto.auditoria.AuditoriaResponseDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface AuditoriaService {
    void registrarEvento(String correoUsuario, String accion, String descripcion, String ip);
    Page<AuditoriaResponseDTO> listarEventos(Pageable pageable);
    Page<AuditoriaResponseDTO> listarEventosPorUsuario(Long usuarioId, Pageable pageable);
}

File: reactivando-frontend/src/services/auditoriaService.js
import api from './api';

export const auditoriaService = {
  listarEventos: async (page = 0, size = 20) => {
    const response = await api.get(`/auditoria?page=${page}&size=${size}`);
    return response.data;
  },

  listarPorUsuario: async (usuarioId, page = 0, size = 20) => {
    const response = await api.get(`/auditoria/usuario/${usuarioId}?page=${page}&size=${size}`);
    return response.data;
  },
};
